### ISSUE TITLE
Error getting the Metadata URL in WP 4.7

### ISSUE THREAD

**Issue by gfirem** [YahnisElsts/plugin-update-checker#75]
Hi, i face this new problem when update WP to 4.7 version. I debug the update with the debug toolbar, and i get the next response from the request. 

```
Notice: Failed to parse plugin metadata. Try validating your .json file with http://jsonlint.com/ in /home/88120-68612.cloudwaysapps.com/rgckemtmsn/public_html/wp-content/plugins/formidable_copy_action/plugin-update-checker/plugin-update-checker.php on line 1007
Failed to retrieve plugin info from the metadata URL.
```
I test the request directly and work fine, and test the json in http://jsonlint.com/ and is well formated. 

Thanks for your help in advance.

**Comment by YahnisElsts** [comment]
It would be useful to know what response the library gets from the server. Try adding some debug output to the `requestInfo` method in `plugin-update-checker.php`. One way to do that would be to add `var_dump($result);` after this code: 

```php
$result = wp_remote_get(
	$url,
	$options
);
```

Then use the "Check Now" button in the debug toolbar again and see what the output is.

**Comment by gfirem** [comment]
Hi, i made the next request 
`http://www.gfirem.com/update-services/?action=get_metadata&slug=formidable_copy_action`
and the response generate a notice. 
`Notice: Failed to parse plugin metadata. Try validating your .json file with http://jsonlint.com/ in /home/88120-68612.cloudwaysapps.com/rgckemtmsn/public_html/wp-content/plugins/formidable_copy_action/plugin-update-checker/plugin-update-checker.php on line 1007`
The problem is in the function json_decode maybe is empty the $json param. 
```
/**
	 * Create a new instance of PluginInfo from JSON-encoded plugin info 
	 * returned by an external update API.
	 * 
	 * @param string $json Valid JSON string representing plugin info.
	 * @return PluginInfo_3_1|null New instance of PluginInfo, or NULL on error.
	 */
	public static function fromJson($json){
		/** @var StdClass $apiResponse */
		$apiResponse = json_decode($json);
		if ( empty($apiResponse) || !is_object($apiResponse) ){
			trigger_error(
				"Failed to parse plugin metadata. Try validating your .json file with http://jsonlint.com/",
				E_USER_NOTICE
			);
			return null;
		}

```
It only happend when i update to wp 4.7 i test in 4.6 and not happen

**Comment by YahnisElsts** [comment]
Yes, I got that from your first post. What I was asking for is the actual response as received by the library. As in, the data returned by the `wp_remote_get` call in the `requestInfo` method.

**Comment by gfirem** [comment]
Hi, i paste the response.

`{"headers":{},"body":"Connection failed: Access denied for user '*******'@'localhost' (using password: YES)","response":{"code":200,"message":"OK"},"cookies":[],"filename":null,"http_response":{"data":null,"headers":null,"status":null}}`

**Comment by YahnisElsts** [comment]
Thank you. Hmm, that's an interesting result. It looks like a MySQL connection error, but that's not the normal WordPress error template. The update server doesn't use a database, so it also can't be the source of this error. Are you *sure* it's requesting the right URL? Are there any proxies or redirects involved?

**Comment by gfirem** [comment]
No, test yourself. 
 http://www.gfirem.com/update-services/?action=get_metadata&slug=formidable_copy_action 
I think is a problem or change in WP 4.7 that affect the wp_remote_get. If was a problem with the server it dont work with any other version of WP, also neither with the direct connection.

**Comment by YahnisElsts** [comment]
I have tested the update checker on WP 4.7 and 4.8-alpha and it worked fine. I also added the metadata URL you posted to a test plugin and it was able to successfully retrieve and parse update information from the URL.

Of course, there's a chance that there's some kind of a compatibility issue that only manifests in certain environments. If so, you might be on your own since I can't seem to reproduce the issue.

The next thing I'd try is monitoring the requests sent by the update checker using Wireshark or something similar.
